packages/lib/shared/components/errors/UnderlyingAddError.tsx

```diff
@@ -0,0 +1,50 @@
+'use client'
+
+import { AlertProps, Text } from '@chakra-ui/react'
+import { ErrorAlert } from './ErrorAlert'
+import { ApiToken } from '@repo/lib/modules/tokens/token.types'
+import { HumanTokenAmountWithAddress } from '@repo/lib/modules/tokens/token.types'
+import { bn } from '@repo/lib/shared/utils/numbers'
+
+type Props = AlertProps & {
+  validTokens: ApiToken[]
+  humanAmountsIn: HumanTokenAmountWithAddress[]
+}
+
+/**
+ * if user attempts to add underlying token amount greater than max deposit of erc4626 vault
+ */
+export function UnderlyingAddError({ humanAmountsIn, validTokens }: Props) {
+  const humanUnderlyingAmountsIn = humanAmountsIn.filter(humanAmount =>
+    validTokens.some(
+      token =>
+        token.address === humanAmount.tokenAddress &&
+        token.wrappedToken &&
+        token.useUnderlyingForAddRemove
+    )
+  )
+
+  const errorMessages = humanUnderlyingAmountsIn
```

Comment:
maybe add the token address as a key here so you don't need to dump the whole error msg as a key on ln 42?